jujugui/static/gui/src/app/app.js

```diff
@@ -1366,23 +1363,23 @@ YUI.add('juju-gui', function(Y) {
             this.storeUser('jem', null, true);
           }
 
-          var envData = this._pickEnv(envList);
-          this.set('environmentList', envList);
+          var modelData = this._pickModel(modelList);
+          this.set('environmentList', modelList);
 
           // XXX frankban: we should try to connect to all the addresses in
           // parallel instead of assuming private addresses must be excluded.
           // The same logic will be then reused for handling HA controllers.
-          var address = window.juju.chooseAddress(envData['host-ports']);
+          var address = window.juju.chooseAddress(modelData.hostPorts);
           if (address === null) {
             console.error(
               'no valid controller address returned by JEM:',
-              envData['host-ports']);
+              modelData.hostPorts);
             return;
           }
           var hostAndPort = address.split(':');
           socketUrl = this.createSocketURL(
-            hostAndPort[0], hostAndPort[1], envData.uuid);
-          callback.call(this, socketUrl, envData.user, envData.password);
+            hostAndPort[0], hostAndPort[1], modelData.uuid);
+          callback.call(this, socketUrl, modelData.user, modelData.password);
```

Comment:
The new Juju 2 api doesn't return username and password - an additional request has to be made to fetch them.